test/sequence/token_refresh_sequence_test.rb

```diff
@@ -7,82 +7,162 @@
-                                                     initiate_login_uri: 'http://localhost:4567/launch',
-                                                     redirect_uris: 'http://localhost:4567/redirect',
-                                                     scopes: 'launch openid patient/*.* profile',
-                                                     refresh_token: refresh_token)
+    @instance = Inferno::Models::TestingInstance.new(
+      url: 'http://www.example.com',
+      client_name: 'Inferno',
+      base_url: 'http://localhost:4567',
+      client_endpoint_key: Inferno::SecureRandomBase62.generate(32),
+      client_id: SecureRandom.uuid,
+      selected_module: 'argonaut',
+      oauth_authorize_endpoint: 'http://oauth_reg.example.com/authorize',
+      oauth_token_endpoint: 'http://oauth_reg.example.com/token',
+      initiate_login_uri: 'http://localhost:4567/launch',
+      redirect_uris: 'http://localhost:4567/redirect',
+      scopes: 'launch/patient online_access openid profile launch user/*.* patient/*.*',
+      refresh_token: refresh_token
+    )
 
     @instance.save! # this is for convenience.  we could rewrite to ensure nothing gets saved within tests.
     client = FHIR::Client.new(@instance.url)
     client.use_dstu2
     client.default_json
     @sequence = Inferno::Sequence::TokenRefreshSequence.new(@instance, client, true)
     @standalone_token_exchange = load_json_fixture(:standalone_token_exchange)
+    @confidential_client_secret = SecureRandom.uuid
   end
 
-  def all_pass(confidential)
+  def setup_mocks(failure_mode = nil)
     WebMock.reset!
-    if confidential
-      # Responses Must Contain Authorization Header
-      stub_request(:post, @instance.oauth_token_endpoint)
-        .with(headers: { 'Content-Type' => 'application/x-www-form-urlencoded',
-                         'Authorization' =>
-                              "Basic #{Base64.strict_encode64(@instance.client_id +
-                                                                  ':' +
-                                                                  @instance.client_secret)}" },
-              body: { 'grant_type' => 'refresh_token',
-                      'refresh_token' => @instance.refresh_token })
-        .to_return(status: 200,
-                   body: @standalone_token_exchange.to_json,
-                   headers: { content_type: 'application/json; charset=UTF-8',
-                              cache_control: 'no-store',
-                              pragma: 'no-cache' })
-
-      # Responses must NOT contain client_id in the body or the client secret in any situation
-      stub_request(:post, @instance.oauth_token_endpoint)
-        .with(body: /client_id|client_secret/)
-        .to_return(status: 401)
+
+    headers = {
+      'Content-Type' => 'application/x-www-form-urlencoded'
+    }
+    body = {
+      'grant_type' => 'refresh_token',
+      'refresh_token' => @instance.refresh_token
+    }
+
+    body_with_scope = {
```

Comment:
```
body_with_scope = body.merge('scope' => @instance.scopes)
```

Reply:
I figured there would be a better way to do this.  Changed.